src/main/java/com/ngs/stash/externalhooks/hook/ExternalPreReceiveHook.java

```diff
@@ -166,33 +164,37 @@ public int runExternalHooks(
         output.close();
 
         boolean trimmed = false;
-        if (hookResponse != null) {
-            int data;
-            int count = 0;
-            while ((data = input.read()) >= 0) {
-                if (count >= 65000) {
-                    if (!trimmed) {
-                        hookResponse.err().
-                            print("\n");
-                        hookResponse.err().
-                            print("Hook response exceeds 65K length limit.\n");
-                        hookResponse.err().
-                            print("Further output will be trimmed.\n");
-                        trimmed = true;
-                    }
-                    continue;
+        int data;
+        int count = 0;
+        StringBuilder builder = new StringBuilder();
+        while ((data = input.read()) >= 0) {
+            if (count >= 65000) {
+                if (!trimmed) {
+                	builder.
+                        append("\n");
+                	builder.
+                        append("Hook response exceeds 65K length limit.\n");
+                	builder.
+                		append("Further output will be trimmed.\n");
+                    trimmed = true;
                 }
+                continue;
+            }
 
-                String charToWrite = Character.toString((char)data);
-
-                count += charToWrite.getBytes("utf-8").length;
+            String charToWrite = Character.toString((char)data);
 
-                hookResponse.err().print(charToWrite);
-            }
+            count += charToWrite.getBytes("utf-8").length;
 
+            builder.append(charToWrite);
+        }
+        
+        int result = process.waitFor();
+        
+        if (result == 0) {
+        	return RepositoryHookResult.accepted();
+        } else {
```

Comment:
This `else` is unnecessary. The `if` above it `return`s, so you could just write:

```java
if (result == 0) {
    return RepositoryHookResult.accepted();
}
return RepositoryHookResult.rejected(summaryMessage, builder.toString());
```